src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -139,7 +133,7 @@ private String decrypt(final String key, final String value) {
         try {
             return getStorageHelper().decrypt(value);
         } catch (GeneralSecurityException | IOException e) {
-            Logger.e(TAG, "Decryption failure", "", ADALError.ENCRYPTION_FAILED, e);
+            Logger.e(TAG, "Decryption failure", "", ADALError.DECRYPTION_FAILED, e);
```

Comment:
If you'd like to get telemetry how frequently this error occurs use ClientAnalytics to log this event and information that might help you to investigate what happened